ocs_ci/utility/service.py

```diff
@@ -0,0 +1,242 @@
+            )
+
+    def kill(self, node, timeout):
+        """
+        Kill the given service using systemctl.
+
+        Args:
+            node (object): Node objects
+            timeout (int): time in seconds to wait for service to be stopped.
+
+        """
+        nodeip = self.nodes[node.name]
+        cmd = f"ssh core@{nodeip} sudo systemctl kill {self.service_name}.service"
+        result = exec_cmd(cmd)
+        ret = TimeoutSampler(
+            timeout=timeout,
+            sleep=3,
+            func=self.verify_service,
+            node=node,
+            action=INACTIVE,
+        )
+        logger.info(f"Result of kill of service {self.service_name} is {result}-{ret}")
+
+    def restart(self, node, timeout):
+        """
+        Restart the given service using systemctl.
+
+        Args:
+            node (object): Node objects
+            timeout (int): time in seconds to wait for service to be started.
+
+        """
+        nodeip = self.nodes[node.name]
+        cmd = f"ssh core@{nodeip} sudo systemctl restart {self.service_name}.service"
+        result = exec_cmd(cmd)
+        ret = TimeoutSampler(
+            timeout=timeout,
+            sleep=3,
+            func=self.verify_service,
+            node=node,
+            action=ACTIVE,
+        )
+        logger.info(
+            f"Result of restart of service {self.service_name} is {result}-{ret}"
+        )
+
+    def status(self, node, timeout):
+        """
+        Get the status of the given service using systemctl.
+
+        Args:
+            node (object): Node objects
+            timeout (int): Future use.
+
+        Returns:
+            (string): 'active' or 'inactive' or 'failed', etc.
+        """
+        nodeip = self.nodes[node.name]
+        cmd = f"ssh core@{nodeip} sudo systemctl status {self.service_name}.service"
+        result = exec_cmd(cmd)
+        logger.info(f"Result of status of service {self.service_name} is {result}")
+        return result.stdout.lower().rstrip()
+
+
+class KubeletService(Service):
+    """
+    Kubelet Service class
+
+    The purpose of this class is to extend Service class to provide stop/start/restart etc operations on
+    kubelet service. Since kubelet service stop and start operations involve changing status of OCP node
+    objects, this class verifies the same.
+    """
+
+    def __init__(self):
+        """
+        Class Initialization.
+
+        Initialize the service with kubelet service name.
+        """
+        super(KubeletService, self).__init__("kubelet")
+
+    def stop(self, node, timeout):
+        """
+        Stop the kubelet service using parent service class. After that, ensures the corresponding OCP node
+        moves to NotReady state.
+
+        Args:
+            node (object): Node objects
+            timeout (int): time in seconds to wait for service to stop.
+
+        """
+        super().stop(node, timeout)
```

Comment:
The implementation looks reversed,  we might actually not need either parent or child class, since you are doing heavy lifting in parent and waiting in child.  I will approve the PR not to hold it further.

Reply:
I will revisit this in the next iteration. I feel that parent class should do as much in a generic way and child class should do whatever is specific to themselves. I will initiate a discussion when we get time to revisit this design.